src/servers/middlewares/wallet-id.ts

```diff
@@ -1,29 +1,54 @@
+import { GraphQLResolveInfo, GraphQLFieldResolver } from "graphql"
+
 import { Accounts } from "@app"
 import { mapError } from "@graphql/error-map"
 
-/* eslint @typescript-eslint/ban-ts-comment: "off" */
-// @ts-ignore-next-line no-implicit-any error
-const validateWalletId = async (resolve, parent, args, context, info) => {
-  const { walletId } = args.input || args || {}
+type InputArgs = Record<"input", Record<string, unknown>>
+
+const validateWalletId = async (
+  resolve: GraphQLFieldResolver<unknown, GraphQLContext | GraphQLContextForUser>,
+  parent: unknown,
+  args: unknown,
+  context: GraphQLContext | GraphQLContextForUser,
+  info: GraphQLResolveInfo,
+) => {
+  const { walletId } = (args as InputArgs).input || args || {}
   if (!walletId) return new Error("Invalid wallet")
   if (walletId instanceof Error) return walletId
 
-  const hasPermissions = await Accounts.hasPermissions(context.domainUser.id, walletId)
+  if (!context.domainUser) {
+    return new Error("Invalid wallet")
+  }
+
+  const hasPermissions = await Accounts.hasPermissions(
+    context.domainUser.id,
+    walletId as WalletId,
```

Comment:
Just a note-to-self that this type is properly applied in the `scalar/wallet-id.ts` scalar object and that this casting will likely be removed later on once the args are properly types here